### ISSUE TITLE
Documentation of l3 color concepts and interfaces

### ISSUE THREAD

**Issue by car222222** [latex3/latex3#746]
This issue contains suggestions for documentation related to l3color itself and any higher-level interfaces that may be built thereon.

Content to follow shortly.

**Comment by Leotuashutosh** [comment]
Hello! I would like to work on this issue if it is still open

**Comment by car222222** [comment]
@Leotuashutosh  Yes, it has just opened so please add your comments, ideas and thougts.

I shall post some material for discussion here later today.

**Comment by car222222** [comment]
PREAMBLE

These paragraphs are not intended to be a single complete  and finalised product, just some possible wordings that highlight and (I hope) explain some of the concepts.  They will need a lot of amendment to link to the l3 documentation and to the details and syntax of various interfaces to be provided at different levels.  They also need careful ‘fact-checking’!

﻿I wrote them because I think that there will be sensible to clarify some of the ideas and terminology and distinctions and to highlight important distinctions between types of models and color spaces. 

I am not claiming that I have explained everything or done it optimally, so let us discuss these ideas further!

Note that there are (almost) ‘no spots on this’:-).

—————
IDEAS 

Note: 
The term ‘mixing colors’, as used here, refers only to quite simple operations on colors that make sense when using traditional printing technologies, and that can be equally applied to ‘screen colors’.  
Thus it does not cover concepts such as the many sophisticated types of ‘blending’ that are described, for example, in this CSS specification document: 
  Compositing and Blending Level 1   at   http://www.w3.org/TR/compositing-1/ 

**Named color**: means a LaTeX name, pretty much as in xcolor.  
For these. tinting/mixing/‘everything‘ are simply calculations done on the LaTeX side. Included here is all the support for tex-related ‘color name sets’ such as those from dvips and, maybe, some other name sets (e.g., as in xcolor).   

Much of what is contained in typical LaTeX documents needs no more than these ‘named colors’, together with the tinting and mixing facilities provided for them by, for example, xcolor.

But many professional-level technologies for output such as ‘print on paper’ need more complex software models to specify colors and also (in some cases) how to print them. 

Some (most?) of these technologies are supported by the PDF standard.  To exploit these features in PDF, some extra information needs to be added from LaTeX.

In the simplest cases this extra information is just to add a ‘PDF name’ for the color (or, more precisely the ‘color sosce’).  But this (PDF) name needs to be one that is recognised by the target consumer of the PDF (typically, software that controls a particular technology for ‘printing in papers) and, in some cases, this name also explicitly identifies a file that contains information about the color (space).

The simplest technology of this type requires as input only the ‘recognised name’ of a single ‘separation color’, often called a ‘spot color’, that is recognised by some consumer (printing) software — this works because such software can use just this name to access the information it needs to produce the required tint of this color in a ‘separation’.  In typical uses of this type of color (space), this will enable the production of a separate printing plate (or film), or the use of an ‘extra process color’.  The color spaces for these single, named, colors are, in PDF, called Separation spaces and it is necessary to set up one Separation space for one for each such color that is used in a document.  Information about each separation that is used in a document must also be added to the pdf file (in a suitable resource dictionary).

For these Separation color spaces, LaTeX provides a new color model, as follows:
**Separation color**: the major property of these is that they must have a PDF name, one that is ‘known’ (to pdf consumers) and this ‘PDF name‘ will ‘appear in a resource dictionary in the pdf file’, together possibly with information about the semantics of tinting (and about mixing, see below), etc.  

Each PDF Separation color (space) also requires a fall-back definition in some well-defined ‘basic color space‘ but (importantly), this fall-back definition will _not_ be used by consumers (e.g., printing) applications that ‘recognise’ the PDF name as they will use this name alone to access all the information they need to produce tints of this color.
Since these color spaces are closely associated with traditional printing with ink, by far the most common basic color space used with them is CMYK, but others, such as RGB  or an ICC spa e can be used.   This fall-back definition includes also a function that maps each valid (in-range) value of the input ‘tint parameter’ to an exact value in the specified ‘basic color space’.

Mixing of Separation colors involves setting up a **DeviceN color space** (and, in general, defining a ‘mixing function and mixing hints‘); a similar but simpler set-up is needed for tints and tinting.  [This part needs expanding.]

**Comment by car222222** [comment]
NOTE: although this is (correctly) labelled ‘documentation’ it also contains language and concepts that may influence commands and their implementation.

**Comment by car222222** [comment]
Update: much of this still needs to be added, together with more recent ideas, e.g. about further 'reserved' color names.

**Comment by josephwright** [comment]
Honestly I think this is one for you @car222222: I am not really sure what is needed.

**Comment by car222222** [comment]
OK, I shall try get around to doing something on this.  Nothing urgent I think.
